assayist/processor/utils.py

```diff
@@ -61,6 +61,16 @@ def download_build_data(build_identifier, output_dir='/metadata'):
     assert_command('koji')
     koji = get_koji_session()
     build = koji.getBuild(build_identifier)
+    if not build.get('source'):
+        # Sometimes there is no source url on the build but it can be found in the task
+        # request info instead. Try looking there, and if found update the build info
+        # so the analyzers have a nice consistent place to find it.
+        try:
+            build['source'] = get_source_of_build(build)
```

Comment:
Since we do this here, I guess it doesn't make sense to repeat the same call in `download_source`:

```python
source_url = get_source_of_build(build_info)
```
In `download_source`, we should simply raise `BuildSourceNotFound` if there is no `build['source']` to keep the current behavior.

That also means that we will never encounter a situation when the fake URL would be needed, right?

Reply:
I've taken your suggestion and re-worked the commit, also as per our conversations on irc.